### PR TITLE
feat: Add hardhat fork

### PR DESCRIPTION
**Pull request by aldoromo88**
## Description of changes

This PR intention is to add hardhat in order to use its fork feature as requested in https://github.com/pancakeswap/pancake-frontend/issues/537

## Changes

 - Add hardhat as dev dependency
 - Add script to run hardhat fork & start app
 - Limit how many farms and pools are loaded for faster development

## Steps to self test
  - Remove `Binance Smart Chain Mainnet` from metamask to avoid conflicts with ChainId
  - Run `npm start-fork`
  - Add hardhat fork network to metamask
image

  - Use app as you normally would do
 
## Test done
 - Swap some BNB for BUSD
 - Add liquidity BNB-BUSD
 - Stack BNB-BUSD-LP
https://recordit.co/xBEYptybyB

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  networks: {
+    hardhat: {
+      chainId: 56,
+      forking: {
+        url: `https://bsc.getblock.io/mainnet/?api_key=${process.env.GETBLOCK_APY_KEY}`,
```
Comment (aldoromo88) [inline hardhat.config.js:12]: Make sure to have your own key exposed as `GETBLOCK_APY_KEY` env var

File: src/config/constants/farms.ts [hunk 2]
```diff
@@ -4385,4 +4385,4 @@ const farms: FarmConfig[] = [
   },
 ]
 
-export default farms
+export default process.env.REACT_APP_USING_FORK === 'true' ? farms.splice(0, 5) : farms
```
Comment (aldoromo88) [inline src/config/constants/farms.ts:4388]: This is not strictly necessary, but I was facing timeout issues on hardhat due to the high amount of farms/pools

File: package.json [hunk 3]
```diff
@@ -5,6 +5,8 @@
   "scripts": {
     "analyze": "source-map-explorer 'build/static/js/*.js'",
     "start": "react-scripts start",
+    "fork": "hardhat node --show-stack-traces --emoji",
+    "start-fork": "npm run fork & REACT_APP_USING_FORK=true npm run start",
```
Comment (aldoromo88) [inline package.json:9]: Please note that this script is injecting `REACT_APP_USING_FORK` env var

**Comment by hachiojidev** [comment]
@aldoromo88 Thank you for submitting this. We will check this out whenever we can squeeze in some time.

**Pull request closed without merging**